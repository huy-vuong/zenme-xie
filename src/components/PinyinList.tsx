import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import styles from 'zenme-xie/components/PinyinList.module.scss';

export default function PinyinList({ id, pinyinList }: PinyinListProp) {
  return (
    <div className={styles.pinyinList}>
      {_.initial(
        pinyinList?.flatMap(
          (pinyin, pinyinIndex) => [
            <code
              id={`${id}-${pinyinIndex}`}
              className={styles.pinyin}
              key={`${id}-${pinyinIndex}`}
            >
              {pinyin}
            </code>,
            <span
              className={styles.pinyinSeparator}
              key={`${id}-separator-${pinyinIndex}`}
            >
              &nbsp;â€¢&nbsp;
            </span>,
          ],
          1
        )
      )}
    </div>
  );
}

interface PinyinListProp {
  id: string;
  pinyinList?: Array<string>;
}

PinyinList.propTypes = {
  id: PropTypes.string.isRequired,
  pinyinList: PropTypes.arrayOf(PropTypes.string.isRequired),
};
