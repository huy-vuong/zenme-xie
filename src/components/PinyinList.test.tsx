import { render } from '@testing-library/react';
import React from 'react';
import PinyinList from 'zenme-xie/components/PinyinList';

test('renders each of the given pinyin', () => {
  const pinyinList = ['māo', 'miāo'];
  const { getByText, asFragment } = render(
    <PinyinList id="test" pinyinList={pinyinList} />
  );
  pinyinList.forEach((text) => getByText(text));
  expect(asFragment()).toMatchSnapshot();
});

test('renders nothing given no pinyin', () => {
  const { asFragment } = render(<PinyinList id="test" pinyinList={[]} />);
  expect(asFragment()).toMatchSnapshot();
});
