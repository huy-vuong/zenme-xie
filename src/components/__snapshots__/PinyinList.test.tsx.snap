// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders each of the given pinyin 1`] = `
<DocumentFragment>
  <div
    class="pinyinList"
  >
    <code
      class="pinyin"
      id="test-0"
    >
      māo
    </code>
    <code
      class="pinyin"
      id="test-1"
    >
      miāo
    </code>
  </div>
</DocumentFragment>
`;

exports[`renders nothing given no pinyin 1`] = `
<DocumentFragment>
  <div
    class="pinyinList"
  />
</DocumentFragment>
`;
